libs/designsystem/menu/src/menu.component.ts

```diff
@@ -75,33 +83,228 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
+      case ' ':
+      case 'Enter':
+      case 'ArrowDown':
+        this.focusedIndex = 0;
+        this.floatingMenu.show();
+        this.focusItem();
+        this.preventFurtherPropagation(event);
+        break;
+      case 'ArrowUp':
+        this.focusedIndex = this.kirbyItems.length - 1;
+        this.floatingMenu.show();
+        this.focusItem();
+        this.preventFurtherPropagation(event);
+        break;
+    }
+  }
+
+  private isPrintableCharacter(key: string) {
+    return key.length === 1 && key.match(/\S/);
+  }
+
+  private handleKeyDownForOpenedMenu(event: KeyboardEvent) {
+    const key = event.key;
+
+    switch (key) {
+      case 'ArrowDown':
+        if (this.focusedIndex === this.kirbyItems.length - 1) {
+          this.focusedIndex = 0;
+        } else {
+          this.focusedIndex++;
+        }
+        this.focusItem();
+        this.preventFurtherPropagation(event);
+        break;
+      case 'ArrowUp':
+        if (this.focusedIndex === 0) {
+          this.focusedIndex = this.kirbyItems.length - 1;
+        } else {
+          this.focusedIndex--;
+        }
+        this.focusItem();
+        this.preventFurtherPropagation(event);
+        break;
+      case 'Home': {
+        if (this.focusedIndex > 0) {
+          this.focusedIndex = 0;
+          this.focusItem();
+        }
+        this.preventFurtherPropagation(event);
+        break;
+      }
+      case 'End': {
+        if (this.focusedIndex < this.kirbyItems.length - 1) {
+          this.focusedIndex = this.kirbyItems.length - 1;
+          this.focusItem();
+        }
+        this.preventFurtherPropagation(event);
+        break;
+      }
+      case 'Escape':
+        if (this.closeOnEscapeKey) {
+          this.floatingMenu.hide();
+        }
+        this.preventFurtherPropagation(event);
+        break;
+      case 'Tab':
+        this.floatingMenu.hide();
+        break;
+      default: {
+        if (this.isPrintableCharacter(key)) {
+          const foundItemIndex = this.getIndexByFirstCharacter(
+            key,
+            this.kirbyItems.map((item) => item.nativeElement.innerText)
+          );
+          if (foundItemIndex > -1) {
+            this.focusedIndex = foundItemIndex;
+            this.focusItem();
+          }
+          this.preventFurtherPropagation(event);
+        }
+      }
+    }
+  }
+
+  private getIndexByFirstCharacter(char: string, words: string[]): number {
+    if (char.length > 1) {
+      return;
+    }
+
+    char = char.toLowerCase();
+    let foundItemIndex = -1;
+    let startIndex = this.focusedIndex + 1;
+
+    if (startIndex !== words.length) {
+      const spliced = words.splice(startIndex);
+      foundItemIndex = spliced.findIndex((word) => word.toLowerCase().startsWith(char));
+    }
+
+    if (foundItemIndex === -1) {
+      startIndex = 0;
+      foundItemIndex = words.findIndex((word) => word.toLowerCase().startsWith(char));
+    }
+    return startIndex + foundItemIndex;
+  }
```

Comment:
Maybe we can simplify this a bit? I've tried to come up with a more readable version, it's not perfect, but let's go through it together 🤞🏻 
```suggestion
  private getItemIndexByFirstCharacter(character: string): number {
    if (character.length > 1) {
      return;
    }
    
    const labelsFirstChar = this.kirbyItems.map((item) =>
      item.nativeElement.innerText[0]?.toLowerCase()
    );
    const charToMatch = character.toLowerCase();
    const findIndexFirstChar = (startIndex: number) =>
      labelsFirstChar.findIndex(
        (firstChar, index) => index >= startIndex && firstChar === charToMatch
      );

    let foundIndex = -1;
    if (this.focusedIndex > -1) {
      // Search from current focused item:
      const startIndex = this.focusedIndex + 1;
      foundIndex = findIndexFirstChar(startIndex);
    }
    if (foundIndex === -1) {
      // If not found in remaining items, search all items:
      const startIndex = 0;
      foundIndex = findIndexFirstChar(startIndex);
    }

    return foundIndex;
  }
```

Reply:
I like this way of writing it, definitely more readable  - however I would suggest to have the array as parameter to the method and not refer to kirbyItems from within. This way it stays generic, and will be easier to refactor out of menu.component to a static util class which can later be used in other placed where similar behavior is needed (dropdown for instance)